libres/lib/enkf/enkf_fs.cpp

```diff
@@ -226,10 +226,11 @@ enkf_fs_type *enkf_fs_alloc_empty(const char *mount_point) {
     if (util_try_lockf(fs->lock_file, S_IWUSR + S_IWGRP, &fs->lock_fd)) {
         fs->read_only = false;
```

Comment:
This will leave `fs->read_only` as always false

Reply:
Not if its able to to get a lock on the file

Reply:
I was thinking about line 232 in the old code. It will set `fs->read_only=true`. Generally, what I think you want to do is:

```cpp
fs->read_only = read_only;
if (!util_try_lockf(fs->lock_file, S_IWUSR + S_IWGRP, &fs->lock_fd)) {
        if (!fs->read_only) {
            util_abort("%s: Another program has already opened filesystem "
                       "read-write \n",
                       __func__);
        }
    }
    return fs;
}
```